tests/monotouch-test/Security/CertificateTest.cs

```diff
@@ -629,5 +629,48 @@ public void GenerateKeyPairTest ()
 				Assert.That (attrs.Count, Is.GreaterThan (0), "private/GetAttributes");
 			}
 		}
+
+		[Test]
+		public void CreateRandomKeyWithParametersTests ()
+		{
+#if IOS || TVOS
+			if (!TestRuntime.CheckXcodeVersion (10, 0)) return;
```

Comment:
`TestRuntime.CheckXcodeVersion` checks the *Xcode* version, not the OS version :smile: It will do the right OS version check matching the specified Xcode version depending on which platform it's running on (it was implemented precisely to avoid adding these kinds of #ifs all over the place).

There's also `TestRuntime.AssertXcodeVersion`, which will call `Assert.Ignore (...);` if the version check fails, and the test will be ignored.

This means you can replace the three different checks with a single (non-conditional) line like this:

```csharp
TestRuntime.AssertXcodeVersion (8, 0);
```

and that will do the right check according to which platform it's running on (iOS/tvOS 10, macOS 10.12, watchOS 3.0).